### DISCUSSION TITLE
How is CLR instance id set?

### DISCUSSION THREAD

**Discussion by huoyaoyuan** [dotnet/runtime#91656] [Q&A] [answered]
When implementing eventing for #91400, `GetClrInstanceId` is used.

The only definition I can find is:

https://github.com/dotnet/runtime/blob/808c523fd2127abb53292ff199cb8b01752d0b54/src/coreclr/nativeaot/Runtime/gcenv.h#L80-L88

On Unix it just returns 0. On Windows I can't find any write for `_tls_index`.

**Comment by jkotas** [comment, marked as answer]
`_tls_index` is intrinsic. It is initialized by the Windows OS loader to the index of thread-static block for the given .dll.
